src/main/java/io/fabric8/maven/XMLFormat.java

```diff
@@ -0,0 +1,206 @@
+        } catch (IOException e) {
+            throw new UncheckedIOException("Could not read XML", e);
+        }
+    }
+
+    XMLOutputter createXmlOutputter() {
+        XMLOutputter xmlOutputter = new XMLOutputter();
+        Format format = Format.getRawFormat();
+        format.setIndent(indent);
+        format.setLineSeparator(LineSeparator.UNIX);
+        format.setTextMode(Format.TextMode.valueOf(textMode.name()));
+        if (insertLineBreakBetweenMajorSections) {
+            // Insert line breaks between major sections
+            xmlOutputter.setXMLOutputProcessor(new LineBreakProcessor());
+        }
+        xmlOutputter.setFormat(format);
+        return xmlOutputter;
+    }
+
+    /**
+     * Find the indentation used in the POM file
+     *
+     * @param pom the path to the POM file
+     * @return the indentation used in the POM file
+     */
+    static String findIndentation(Path pom) {
+        try (BufferedReader br = Files.newBufferedReader(pom)) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                int idx = line.indexOf("<");
+                // We don't care about the first line or unindented lines
+                if (idx > 0) {
+                    return line.substring(0, idx);
+                }
+            }
+        } catch (IOException e) {
+            throw new UncheckedIOException("Could not read POM file: " + pom, e);
+        }
+        return XMLFormat.DEFAULT.getIndent();
+    }
+
+    public enum TextMode {
+        /**
+         * Mode for literal text preservation.
+         */
+        PRESERVE,
+
+        /**
+         * Mode for text trimming (left and right trim).
+         */
+        TRIM,
+
+        /**
+         * Mode for text normalization (left and right trim plus internal
+         * whitespace is normalized to a single space.
+         *
+         * @see org.jdom2.Element#getTextNormalize
+         */
+        NORMALIZE,
+
+        /**
+         * Mode for text trimming of content consisting of nothing but
+         * whitespace but otherwise not changing output.
+         */
+        TRIM_FULL_WHITE;
+    }
+
+    /**
+     * Create a new builder
+     *
+     * @return a new builder
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * Create a new builder with the given {@link XMLFormat}
+     *
+     * @param format the format
+     * @return a new builder
+     */
+    public static Builder builder(XMLFormat format) {
+        return new Builder()
+                .indent(format.getIndent())
+                .insertLineBreakBetweenMajorSections(format.isInsertLineBreakBetweenMajorSections())
+                .textMode(format.getTextMode());
+    }
+
+    public static class Builder {
+        private String indent = "  ";
+
+        private boolean insertLineBreakBetweenMajorSections = false;
+
+        private TextMode textMode = TextMode.TRIM;
```

Comment:
Why do you have TRIM by default here, whereas `XmlFormat.DEFAULT`  uses PRESERVE?

Reply:
I noticed that the indent wasn't applied when PRESERVE is used

Reply:
OK.